### PR TITLE
Forbid case where both `suite.rc` and `flow.cylc` files are found in src/run dir

### PR DESCRIPTION
**Pull request by datamel**
Currently if you have both `flow.cylc` and `suite.rc` files in the run directory and cylc play. You get a log message warning that `suite.rc` has been detected and back compatibility mode has been activated. The workflow proceeds to run ignoring the `suite.rc` file and using the `flow.cylc` file.

This could cause some confusion for users and so we should raise an error if both files are detected (discussed UK end), forcing the user to decide which one should be used, rather than assuming either way.
 
Ideally this would be included in upcoming beta release, apologies, I understand this is frozen.

This is a small change with no associated Issue, bit of a quick fix. Looking at the code, I think there is potential for some refactoring but that would be a bigger job for another day.

To test on master:
```
$ tree `~/cylc-src/workflow`
├── flow.cylc
└── suite.rc
$ cylc install
Installed....
```
Then run `cylc play` on newly installed workflow.
Deprecation warnings occur and back compat mode is enabled, `flow.cylc` is used.

On this branch:

Cylc install/reinstall will raise an error.
Cylc play on a workflow containing `flow.cylc` and `suite.rc` will also raise error.

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.py` and
  `conda-environment.yml`.
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] (master branch) I have opened a documentation PR at https://github.com/cylc/cylc-doc/pull/313

### CODE REVIEW

File: cylc/flow/workflow_files.py [hunk 1]
```diff
@@ -1698,6 +1698,24 @@ def get_run_dir_info(
     return relink, run_num, rundir
 
 
+def detect_both_flow_and_suite(path):
+    """Detects if both suite.rc and flow.cylc are in directory.
+    Raises:
+        WorkflowFilesError: If both flow.cylc and suite.rc are in directory
+    """
+    msg = (f"Both {WorkflowFiles.FLOW_FILE} and {WorkflowFiles.SUITE_RC} "
```
Comment (wxtim) [inline cylc/flow/workflow_files.py:1706]: IMO a link to https://cylc.github.io/cylc-doc/latest/html/7-to-8/summary.html#backward-compatibility would be helpful.
I've asked Element whether we can use a link shortener.

Comment (datamel) [reply cylc/flow/workflow_files.py:1706]: Happy to add a link. I shall be making a doc PR to make this change clear to users too.
I personally would prefer not to use link shortener. I think getting the cylc url in there is good, sometimes I think link shorteners look a bit spammy, as always happy to be overruled though!

**Comment by datamel** [comment]
@hjoliver, I have tagged you as a reviewer, more to make you aware of the issue (and confirmation of suitability for freezed 8.0b3). If you are happy with the combo of `flow.cylc` and `suite.rc` being forbidden in run dirs and source dirs, we can get the code reviewed UK end if it would ease your time pressures.

File: cylc/flow/workflow_files.py [hunk 2]
```diff
@@ -1698,6 +1698,24 @@ def get_run_dir_info(
...
+           "files are present in the run directory. Please remove one and"
+           " try again.")
+    if path.resolve().name == WorkflowFiles.SUITE_RC:
+        flow_cylc = path.parent / WorkflowFiles.FLOW_FILE
+        if flow_cylc.is_file() and not flow_cylc.is_symlink():
+            raise WorkflowFilesError(msg)
+    elif (path / WorkflowFiles.SUITE_RC).is_file():
+        flow_cylc = path / WorkflowFiles.FLOW_FILE
+        if flow_cylc.is_file() and not flow_cylc.is_symlink():
+            raise WorkflowFilesError(msg)
```
Comment (wxtim) [inline cylc/flow/workflow_files.py:1716]:
```suggestion
    if path.resolve().name == WorkflowFiles.SUITE_RC:
        flow_cylc = path.parent / WorkflowFiles.FLOW_FILE
    elif (path / WorkflowFiles.SUITE_RC).is_file():
        flow_cylc = path / WorkflowFiles.FLOW_FILE

    if flow_cylc.is_file() and not flow_cylc.is_symlink():
        raise WorkflowFilesError(msg)
```

**Comment by MetRonnie** [comment]
Is there any need for us to continue creating a flow.cylc symlink to suite.rc? I think it would simplify things if we stopped doing that.

File: cylc/flow/workflow_files.py [hunk 3]
```diff
@@ -1744,7 +1762,13 @@ def check_flow_file(
     suite_rc_path = Path(expand_path(path), WorkflowFiles.SUITE_RC)
     if flow_file_path.is_file():
         if not flow_file_path.is_symlink():
-            return flow_file_path
+            if not suite_rc_path.is_file():
+                return flow_file_path
+            raise WorkflowFilesError(
+                f"Both {WorkflowFiles.FLOW_FILE} and "
```
Comment (wxtim) [inline cylc/flow/workflow_files.py:1768]: If you agree about the link, thenh here too.

File: cylc/flow/workflow_files.py (hunk 2 above)
Comment (wxtim) [reply cylc/flow/workflow_files.py:1716]: Discussed on phone: This doesn't work because flow_cylc is only defined inside the first if..else loop.

**Review by wxtim — APPROVED** [review]
- [x] Code looks good.
- [x] Tested as working.

**Comment by datamel** [comment]
> Is there any need for us to continue creating a flow.cylc symlink to suite.rc? I think it would simplify things if we stopped doing that.

I will remove this in follow up issue.

**Comment by hjoliver** [comment]
> Ideally this would be included in upcoming beta release, apologies, I understand this is frozen.

No problem, newly discovered bugs that can be quickly fixed are fair game.

**Review by hjoliver — COMMENTED** [review]
Looks good, tested as working, but could you make a quick test for the `detect_both_flow_files()` function?

(Or could the two separate checks be made to use the same logic?)

**Comment by MetRonnie** [comment]
I have created a question issue for stopping symlinking: #4500

**Comment by datamel** [comment]
> I have created a question issue for stopping symlinking: #4500

Thanks @MetRonnie, I've got a branch with it removed, just waiting for tests to pass and I'll open the PR.

**Review by hjoliver — APPROVED** [review]
All good, thanks @datamel

**Pull request merged**